### ISSUE TITLE
Any plans to allow multiple proxy connectors?

### ISSUE THREAD

**Issue by EugenMayer** [blacklabelops/jira#27]
Du to the nature of service-desk you might have a different customer-domain for your service-desk compared to the domain for your "internal" use of jira.

Handling this requires 2 different connector entries, but right now only one is possible.

Shall we make the service.xml be overwriteable by a custom file - one we mount somewhere and copy over on very container start ( needed in case of codebase updates )? This would open a wide range of flexible solutions for customizing server.xml without you being in charge implementing it bit by bit ( though i really like the ENV way..)

**Comment by EugenMayer** [comment]
I see that you are rather manipulating the server.xml https://github.com/blacklabelops/jira/blob/master/imagescripts/docker-entrypoint.sh#L19 not replacing it - this keeps the compatibility high with future releases but makes it hard to deal with this particular feature.

**Comment by blacklabelops** [comment]
I never used two connectors inside server.xml. I need an example on what your trying to do.

**Comment by EugenMayer** [comment]
sure. Consider you want to access your jira with 2 different domains: jira.company.de for the internal use and support.company.com for the "servicedesk case" - public facing.

That said, we used 2 different connectors for this

```xml
<Connector acceptCount="100" connectionTimeout="20000" disableUploadTimeout="true" enableLookups="false" maxHttpHeaderSize="8192" maxThreads="150" minSpareThreads="25" port="8080" protocol="HTTP/1.1" proxyName="jira.company.de" proxyPort="443" redirectPort="8443" scheme="https" useBodyEncodingForURI="true"/>

<Connector acceptCount="100" connectionTimeout="20000" disableUploadTimeout="true" enableLookups="false" maxHttpHeaderSize="8192" maxThreads="150" minSpareThreads="25" port="18080" protocol="HTTP/1.1" proxyName="support.company.com" proxyPort="443" redirectPort="8443" scheme="https" useBodyEncodingForURI="true"/>
```

This way you have no issues with jira doing that kind of thing. How do you solve this?

**Comment by blacklabelops** [comment]
Looks pretty easy to me. 

- Use xmlstarlet like me to copy the existing connector node. 
- Then use enumerated envs to adjust domain names in both connectors separately at startup. 
- Add another env for specifying the ports.
- Delete and update connectors at each startup (e.g. deleting all connectors but the first one)

Optional workaround:
- Add env for specifying a server.xml -> Stops all adjustment by other envs. server.xml must be specified completely.

**Comment by blacklabelops** [comment]
Will be implemented in #12.
